### ISSUE TITLE
Parse error handling "except" in BigQuery queries

### ISSUE THREAD

**Issue by jml** [alanmcruickshank/sqlfluff#330]
```console
➜ sqlfluff version
0.3.4
```

Config:

```ini
[sqlfluff]
dialect = bigquery
templater = jinja

[sqlfluff:rules]
tab_space_size = 2
```

The following SQL fails with a parse error:

```sql
select
  * replace 'thing' as foo
from some_table
```

```
➜ sqlfluff lint test.sql
== [test.sql] FAIL
L:   2 | P:   4 |  PRS | Found unparsable section: " replace 'thing' as foo"
```

For reference, https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax is the BigQuery syntax.

I'd suggest looking at #329 in conjunction with this.

**Comment by alanmcruickshank** [comment]
Wow, this one was much harder to handle that expected. Has highlighted some bugs deep down in the parser which I'm really glad are now out in the open. PR coming shortly.
